crates/cubecl-core/src/frontend/element/atomic.rs

```diff
@@ -12,28 +12,40 @@ use crate::{
     unexpanded,
 };
 
+/// An atomic type. Represents an shared value that can be operated on atomically.
 pub trait Atomic: Sized + CubeType
 where
     ExpandElement: From<<Self::Primitive as CubeType>::ExpandType>,
     ExpandElement: From<<Self as CubeType>::ExpandType>,
 {
+    /// The numeric primitive represented by the atomic wrapper.
     type Primitive: Numeric;
 
+    /// Load the value of the atomic. This operation is only atomic in `wgpu`, CUDA does not support
+    /// atomic loads.
     #[allow(unused_variables)]
     fn load(pointer: &Self) -> Self::Primitive {
```

Comment:
What is the current fix for this? Can we use atomic add with 0 to simulate an atomic load?

Reply:
I think eventually we could emit inline PTX, or emit a thread fence if it's really necessary. But most of the time atomic loads aren't actually necessary and thread fences are a pretty significant performance degredation so I would go for just the note in the docs for the time being. We can still change the exact semantics later I think. The main reason the method exists is because `wgpu` *only* allows atomic loads for atomics variables and doesn't allow direct access.

Reply:
Now that I've thought about it more and looked at the code, I think inline PTX is actually the way to go. Using inline PTX sucks when you're writing normal code, but I think it actually works fine if we're just compiling to it. I'll try to implement it.

Reply:
I have once again been mislead by the internet. PTX does not in fact support atomic loads, just non-tearing loads/stores. I think that idea about `atomicAdd` is a good one, with `atomicExch` for store since they're identical, the only difference being that `atomicExch` returns the value. Apparently the CUDA compiler will just optimize that out anyways and replace it with the appropriate instruction.